docs/articles/baseplatform/reproducing-yocto-builds.rst

```diff
@@ -0,0 +1,91 @@
+
+To create a local mirror, the following configuration needs to be added to build/conf/local.conf (note that this is not a default configuration):
+
+.. code-block:: none
+
+    BB_GENERATE_MIRROR_TARBALLS = "1"
+
+During the Yocto build process, packages are downloaded and stored in the downloads/ directory (or whatever directory has been specified in build/conf/local.conf) of the Yocto build's root directory. When set to "1", ``BB_GENERATE_MIRROR_TARBALLS`` forces bitbake to create tar-balls for all the sources that are fetched using git. Note that the build must be done from scratch otherwise some source code packages may not be packaged into tar-balls. In other words, the build cache needs to be wiped clean before the build is started. When everything is set up, start the build as usual with bitbake.
+
+
+Saving the repo manifest
+------------------------
+
+The above instructions describe how to create the local mirror. To be able to reproduce a build, the manifest file used must also be saved. In practice, this means that the manifest file must be pushed to a git repository because the repo tool used when initializing build directories only fetches manifests using git.
+If the manifest only specifies specific commits for each project then the manifest file can be pushed directly to a release branch. If the manifest on the other hand points to some branch or tag of one or more projects, then a manifest with the exact commits used needs to be generated. This can be done with the repo tool from the root of the Yocto build directory after the repo sync command has been issued:
+
+.. code-block:: bash
+
+    $ repo manifest -o - -r >> example-manifest.xml
+
+The generated file is then pushed to the relevant release branch of the manifest repository.
+
+
+Reproducing the Yocto build
+---------------------------
+
+The instructions in this section describes how a Yocto build may be reproduced given that the instructions in the above sections have been followed.
+The download directory previously populated should be made available to the build environment to be used during build reproduction, e.g. by copying the download directory from the archive to a source directory in the root directory of the new Yocto build. The build directory should now look like this:
+.. code-block:: none
+
+    .
+    └── source-mirror/
+
+When initializing the build directory, use a manifest that has been previously associated with the release to be reproduced:
+
+.. code-block:: bash
+
+    $ repo init -u ssh://git@git.example.net/example-group/example-manifest-repository.git -b releases -m example-manifest.xml
+    $ repo sync
+
+Then follow the instructions to initialize the build directory.
+
+The build directory should look like this:
+
+.. code-block:: none
+
+    .
+    ├── build/
+    ├── source-mirror/
+    └── sources/
+
+Once the recipes have been fetched and the environment set up, add the following lines to build/conf/local.conf:
```

Comment:
Once one gets to here, the recipes have already been fetched and the environment set up, no?